docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
 # Developer Guide
 
 ## Design & implementation
+<!-- @@author iamchenjiajun -->
+### Architecture
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+##### Architecture Diagram
+Below is an architecture diagram of termiNus.
+
+![ArchitectureDiagram](./images/ArchitectureDiagram.png)
+
+- `Duke` is the main object of the program and handles all the logic and models related to the program.
+- `Ui` is the main object that provides an interface between `Duke` and the user.
+- `Command` represents a command that is provided by the user. `Ui` reads the command and it is sent to `Parser` to create a new `Command` object.
+- `Duke` executes the command and shows the output to the user using `Ui`.
+- `Command` object modifies the state of `ItemList`, which consists of multiple lists for different types of `Items`.
+- `Storage` takes the state of `ItemList` and stores it to file.
```

Comment:
Good job, I like how the architecture diagram is used here. Easy to understand

Reply:
I think its a good idea to have this diagram first but it may be a bit confusing to those looking at it for the first time. Maybe adding some colour to help differentiate the different parts would be good? I also noticed that you did not describe all the items in the diagram in the explanation, if these are the main components of your architecture then maybe you can narrow it down to this as well?